### ISSUE TITLE
uniq clause breaks default scope with SQL function

### ISSUE THREAD

**Issue by dholdren** [globalize/globalize#519]
Looks like this PR https://github.com/globalize/globalize/pull/489 added `.uniq`.  But if my default scope is `order("LOWER(name)")` or similar, then `find_by` will fail with: 
`PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list`

**Comment by dalpo** [comment]
@dholdren Could you try if the issue is already present with the latest master?
I pushed this commit that (in thery) should fixed the issue: https://github.com/globalize/globalize/commit/3610db4cb2447b3ea25d64bf0682dc9704623e47

**Comment by shioyama** [comment]
This needs to be fixed before we can release a new version (#543).

**Comment by shioyama** [comment]
@dalpo Could you maybe just add a spec that checks your fix works? Or maybe there is one and I missed it...

**Comment by dalpo** [comment]
@shioyama I encountered the same issue in a my project using postgres... In the next days I would like to start working on. As first step I'm going to create a new PR with a failing spec so we can eventually discuss how to fix it.

**Comment by dholdren** [comment]
unfortunately, that did not fix my bug, here's a rough approximation of my test case and the generated SQL:

```ruby

**Comment by dholdren** [comment]
I do have I18n fallbacks enabled, which is why there's multiple locales in the query I believe

**Comment by dalpo** [comment]
@dholdren I started working on a new PR: https://github.com/globalize/globalize/pull/571
I hope to fix it soon.

**Comment by shioyama** [comment]
@dholdren We can't support direct SQL on translated attributes, so `order("LOWER(name)")` will never work. You will need to directly reference the translations table.

**Comment by Istanful** [comment]
@shioyama 

This solution solves the problem for me when running rails 4.2.10. 

```ruby
module Globalize
  module ActiveRecord
    module ClassMethods
      if ::activerecord::version::string < "5.0.0"
        def self.columns
          super.reject do |column|
            translated_attribute_names.map(&:to_s).include? column.name
          end
        end
      end
    end
  end
end
```
By rejecting the translated attributes from the columns array activerecord won't try to query the corresponding methods. There is already a [similar solution](https://github.com/globalize/globalize/blob/e37c471775d196cd4318e61954572c300c015467/lib/globalize/active_record/class_methods.rb#L6-L14) present in the codebase that modifies the `columns_hash` instead. However in rails 4.2.10 the `columns_hash` is generated from the `columns` method.

I hope this might be helpful!
